testsuite/MDAnalysisTests/coordinates/test_xdr.py

```diff
@@ -25,257 +25,209 @@
 warnings.simplefilter('always')
 
 
-class _XDRReader_Sub(TestCase):
-    __test__ = False
-
-    def setUp(self):
-        """
-        grab values from selected atoms from full solvated traj,
-        later compare to using 'sub'
-        """
+class _XDRReader_Sub(object):
+    @pytest.fixture()
+    def atoms(self):
         usol = mda.Universe(PDB_sub_sol, self.XDR_SUB_SOL)
-        atoms = usol.select_atoms("not resname SOL")
-        self.ts = atoms.ts
-        self.sub = atoms.indices
-        # universe from un-solvated protein
-        self.udry = mda.Universe(PDB_sub_dry)
+        return usol.select_atoms("not resname SOL")
 
     def test_load_new_raises_ValueError(self):
         # should fail if we load universe with a trajectory with different
         # number of atoms when NOT using sub, same as before.
-        def load_new_without_sub():
-            self.udry.load_new(self.XDR_SUB_SOL)
-
+        udry = mda.Universe(PDB_sub_dry)
         with pytest.raises(ValueError):
-            load_new_without_sub()
+            udry.load_new(self.XDR_SUB_SOL)
 
-    def test_sub_coordinates(self):
+    def test_sub_coordinates(self, atoms):
         """
         load solvated trajectory into universe with unsolvated protein.
         """
-        self.udry.load_new(self.XDR_SUB_SOL, sub=self.sub)
-        ts = self.udry.atoms.ts
-        assert_timestep_almost_equal(ts, self.ts)
+        udry = mda.Universe(PDB_sub_dry)
+        udry.load_new(self.XDR_SUB_SOL, sub=atoms.indices)
+        ts = udry.atoms.ts
+        assert_timestep_almost_equal(ts, atoms.ts)
 
 
 class TestTRRReader_Sub(_XDRReader_Sub):
-    __test__ = True
     XDR_SUB_SOL = TRR_sub_sol
 
 
 class TestXTCReader_Sub(_XDRReader_Sub):
-    __test__ = True
     XDR_SUB_SOL = XTC_sub_sol
 
 
-class _GromacsReader(TestCase):
-    __test__ = False
+class _GromacsReader(object):
     # This base class assumes same lengths and dt for XTC and TRR test cases!
     filename = None
-    ref_unitcell = np.array([80.017, 80.017, 80.017, 60., 60., 90.],
-                            dtype=np.float32)
+    ref_unitcell = np.array(
+        [80.017, 80.017, 80.017, 60., 60., 90.], dtype=np.float32)
     # computed with Gromacs: 362.26999999999998 nm**3 * 1000 A**3/nm**3
     ref_volume = 362270.0
 
-    def setUp(self):
-        # loading from GRO is 4x faster than the PDB reader
-        self.universe = mda.Universe(GRO, self.filename, convert_units=True)
-        self.trajectory = self.universe.trajectory
-        self.prec = 3
-        self.ts = self.universe.coord
-        # dummy output file
-        ext = os.path.splitext(self.filename)[1]
-        self.tmpdir = tempdir.TempDir()
-        self.outfile = self.tmpdir.name + '/xdr-reader-test' + ext
-
-    def tearDown(self):
-        try:
-            os.unlink(self.outfile)
-        except:
-            pass
-        del self.universe
-        del self.tmpdir
+    prec = 3
 
+    @pytest.fixture(scope='class')
```

Comment:
Is it safe to have only one universe per class, given that various tests rewind or use next? 

Or does parallel pytest create one class-scoped fixture per process so that there is no parallel collision?

I assume the latter... just asking for clarification.

Reply:
There  is an experimental grouping option so that every test in a class runs on the same process. This isn't guaranteed currently. But I haven't found any errors so far. I'm also not sure when fixtures are evaluated. This is because all tests are scanned by all workers before xdist runs all tests. This likely means that each process gets it's own copy and tmp directory.